src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -220,7 +220,14 @@ private Task StartAsyncCore()
             {
                 // Get a connection ID from the server
                 logger.EstablishingConnection(url);
-                using (var request = new HttpRequestMessage(HttpMethod.Options, url))
+                var urlBuilder = new UriBuilder(url);
+                if (!urlBuilder.Path.EndsWith("/"))
+                {
+                    urlBuilder.Path += "/";
+                }
+                urlBuilder.Path += "negotiate";
+
+                using (var request = new HttpRequestMessage(HttpMethod.Post, urlBuilder.Uri))
```

Comment:
does this work fine with queryString? I think we now need a test where the url contains a queryString.

Reply:
We definitely need tests now. I did manually verify that queryString is kept intact